code/go/0chain.net/smartcontract/dbs/event/partition.go

```diff
@@ -25,14 +31,39 @@ func (edb *EventDb) dropPartition(round int64, table string) error {
 }
 
 func (edb *EventDb) movePartitionToSlowTableSpace(round int64, table string) error {
-	number := round / edb.settings.PartitionChangePeriod
-	toMove := number - edb.settings.PartitionKeepCount
-	if toMove < 0 {
+	tablespace := edb.dbConfig.Slowtablespace
+
+	var results []TableInfo
+	err := edb.Store.Get().Table("pg_tables").
+							Select("tablename, tablespace").
+							Where("tablename LIKE ?", table+"_%%").
+							Find(&results).Error
+	if err != nil {
+		return err
+	}
+
+	// remove tables that are part of cold storage
+	var hotTables []TableInfo
+	for i, table := range results {
+		if table.Tablespace != tablespace {
+			hotTables = append(hotTables, results[i])
+		}
+	}
+
+	if len(hotTables) < 10 {
+		// move to hot table only if there are more than 10 tables
 		return nil
 	}
 
-	tablespace := edb.dbConfig.Slowtablespace
-	// identify the partition table that needs to be moved to slow partition
-	raw := fmt.Sprintf("ALTER TABLE %v_%v SET TABLESPACE %v", table, toMove, tablespace)
-	return edb.Store.Get().Exec(raw).Error
+	// sort by tablename
+	sort.Slice(hotTables, func(i, j int) bool {
+		return hotTables[i].Tablename > hotTables[j].Tablename
```

Comment:
are you sure table_round will be sorted accurately? round_1 and round_10 will be sorted correctly?